Workflow

andreasegeberg opened the pull request.

Commit:
Updated workflow for Backbone and Dhcp-boot app's

- Github workflow is changed to one workflow per app.
- Renaming of defconfig files to follow Netbox standard.
- More generic code in workflow yml files.
- Added a defconfig file for building on Dagger platform.

Signed-off-by: Andreas Egeberg <andreas.egeberg@gmail.com>

Commit:
Removing testbranch from workflow build action.

Workflow for each projects is building for platform:
- Zero
- Dagger
- Coronet
Testbranch is removed from workflow.

Signed-off-by: Andreas Egeberg <andreas.egeberg@gmail.com>

Commit:
README is updated to reflect platform builds.

Links on https://nightly.link/ should be updated on next build of master.

Signed-off-by: Andreas Egeberg <andreas.egeberg@gmail.com>

troglobit commented on README.md:
```diff
@@ -51,17 +51,19 @@ Once the build is done, your app image is available in
->           be supported are the _Dagger_ and _Zero_ platforms.  For
+
+> **Note:** Currently the nightly builds and defconfigs available here target
+>           the Westermo _Coronet_, _Dagger_ and _Zero_ platforms.  Other platforms planned to will 
```
Itty bitty comment here on line lengths in text files,  I try to keep them <80 chars.  I have Emacs set up to reflow paragraphs for me with Meta-Q, so I have it easy.  The intent is to make it easy on the eyes also when reading text files using less or similar on a small terminal.

troglobit commented on src/backbone/backbone.c:
```diff
@@ -94,9 +94,11 @@ static int backbone_cb_id(struct backbone_id *id)
 static bool backbone_cb_update(void)
 {
-	struct backbone_id new = { 0 }, *id;
+	/* struct backbone_id = {{ 0 }}, *id; */  /* correctly zero-initialized */
```
Just drop this line instead of commenting it out, we have git for file history :)

troglobit commented on src/backbone/backbone.c:
```diff
@@ -94,9 +94,11 @@ static int backbone_cb_id(struct backbone_id *id)
 {
-	struct backbone_id new = { 0 }, *id;
+	/* struct backbone_id = {{ 0 }}, *id; */  /* correctly zero-initialized */
+	struct backbone_id new, *id;
```
We should probably rename all variables named `new` to `new_id` or something.  Just to at least try and avoid clashing with reserved C++ keywords.

troglobit reviewed (changes requested):
Very minor changes needed, just push follow-up commits for each, then you can merge :)

Commit:
Create md5 for dhcp-boot artifact.

Since fawlty is going to download a lot of built binaries
it must be possible to check if the artifact is already downloaded
or if it is a new build that must be downloaded.

Signed-off-by: Andreas Egeberg <andreas.egeberg@westermo.se>

Commit:
Github action creates hash file to identify unique artifacts.

Since fawlty is going to download a lot of prebuilt binaries
it must be possible to check if the artifact is already downloaded
or if there is a new build that must be downloaded.
This hash is not a software signature, only an unique id for the commit.

Signed-off-by: Andreas Egeberg <andreas.egeberg@westermo.se>

andreasegeberg committed:
Update nightly-dhcp-boot.yml

Changed github action

andreasegeberg committed:
Update nightly-dhcp-boot.yml

Tested MCjack123 again

andreasegeberg committed:
Update nightly-dhcp-boot.yml

Every step must have a run or an uses action

Commit:
Generate hash file on release event

Signed-off-by: Andreas Egeberg <andreas.egeberg@westermo.se>

Commit:
Added commit hash to a file.

Signed-off-by: Andreas Egeberg <andreas.egeberg@westermo.se>

Signed-off-by: Andreas Egeberg <andreas.egeberg@westermo.se>

Commit:
separate hash file

Commit:
removed tabs

Commit:
logic error

Commit:
filepath added

Commit:
filepath changed

Commit:
final

Commit:
yaml error

Commit:
Keep paragraph < 80 char for readability.

Signed-off-by: Andreas Egeberg <andreas.egeberg@westermo.se>

Commit:
Removed commented line.

Signed-off-by: Andreas Egeberg <andreas.egeberg@westermo.se>

Commit:
Rename variable new to new_id.

Avoid using reserved C++ keywords for more pragmatic code experince.

Signed-off-by: Andreas Egeberg <andreas.egeberg@westermo.se>

Commit:
Merge branch 'workflow' of https://github.com/westermo/app-demo into workflow
